AssertionError: Unexpected value for field original_item when creating Stock Entry for Manufacture if BOM items have Item operation set

### Information about bug

When a BOM has `Item operation` set on its items and `Transfer Material Against = Job Card`, creating a Stock Entry for Manufacture throws:

```
AssertionError: Unexpected value for field original_item: ['140061.00', 'Trapezoidal Finishing']
```

**Steps to Reproduce:**

1. Create a BOM with `With Operations = 1`, `Transfer Material Against = Job Card`
2. Set `Item operation` on one or more items in the BOM items table (e.g. link `140061.00` to operation `Trapezoidal Finishing`)
3. Create a Work Order from this BOM
4. Complete all Job Cards
5. Try to create the final Manufacture Stock Entry

**Expected:** Stock Entry saves successfully

**Observed:** `AssertionError` on save — Stock Entry cannot be saved

**Root Cause:**

In `erpnext/manufacturing/doctype/bom/bom.py`, `get_bom_items_as_dict()` builds a tuple key when `item.operation` is set:

```python
key = item.item_code
if item.operation:
    key = (item.item_code, item.operation)  # becomes a tuple
```

In `erpnext/stock/doctype/stock_entry/stock_entry.py`, this tuple then gets written into `original_item` which is a Link field expecting a plain string:

```python
for original_item, item in item_dict.items():
    if original_item != item.get("item_code"):
        item["original_item"] = original_item  # stores tuple → crash
```

**Proposed Fix:**

In `stock_entry.py`, add an `isinstance` check:

```python
if isinstance(original_item, str) and original_item != item.get("item_code"):
    item["original_item"] = original_item
```

**ERPNext Version:** 16.9.0
**Frappe Version:** 16.11.0

### Module

manufacturing

### Version

ERPNext Version: 16.9.0
Frappe Version: 16.11.0

### Installation method

None

### Relevant log output / Stack trace / Full Error Message.

```shell
### App Versions

{
	"frappe": "16.11.0",
	"erpnext": "16.9.0",
	"email_delivery_service": "0.0.1",
	"erpnext_germany": "15.23.1",
	"print_designer": "1.6.5",
	"procudi_erpdev": "0.0.1"
}

### Route

Form/Stock Entry/new-stock-entry-zdzopmrpsf

### Traceback

Traceback (most recent call last):
  File "apps/frappe/frappe/app.py", line 121, in application
    response = frappe.api.handle(request)
  File "apps/frappe/frappe/api/__init__.py", line 63, in handle
    data = endpoint(**arguments)
  File "apps/frappe/frappe/api/v1.py", line 40, in handle_rpc_call
    return frappe.handler.handle()
           ~~~~~~~~~~~~~~~~~~~~~^^
  File "apps/frappe/frappe/handler.py", line 53, in handle
    data = execute_cmd(cmd)
  File "apps/frappe/frappe/handler.py", line 86, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
           ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/__init__.py", line 1129, in call
    return fn(*args, **newargs)
  File "apps/frappe/frappe/utils/typing_validations.py", line 36, in wrapper
    return func(*args, **kwargs)
  File "apps/frappe/frappe/desk/form/save.py", line 43, in savedocs
    doc.save()
    ~~~~~~~~^^
  File "apps/frappe/frappe/model/document.py", line 518, in save
    return self._save(*args, **kwargs)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "apps/frappe/frappe/model/document.py", line 540, in _save
    return self.insert()
           ~~~~~~~~~~~^^
  File "apps/frappe/frappe/model/document.py", line 440, in insert
    self._validate_links()
    ~~~~~~~~~~~~~~~~~~~~^^
  File "apps/frappe/frappe/model/document.py", line 1139, in _validate_links
    result = d.get_invalid_links(is_submittable=self.meta.is_submittable)
  File "apps/frappe/frappe/model/base_document.py", line 994, in get_invalid_links
    assert isinstance(docname, str | int), f"Unexpected value for field {df.fieldname}: {docname}"
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^
AssertionError: Unexpected value for field original_item: ['150002.02', 'Laser Engraving']
```

--- a/erpnext/stock/doctype/stock_entry/stock_entry.py
+++ b/erpnext/stock/doctype/stock_entry/stock_entry.py
@@ -2430,7 +2430,7 @@ class StockEntry(StockController, SubcontractingInwardController):
 							self.to_warehouse if self.purpose == "Send to Subcontractor" else ""
 						)
 
-						if original_item != item.get("item_code"):
+						if isinstance(original_item, str) and original_item != item.get("item_code"):
 							item["original_item"] = original_item
 
 					self.add_to_stock_entry_detail(item_dict)

